client/src/components/WatchlistButtons.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import WatchlistIcon from '../assets/watchlist-icon.svg';
+import WatchlistIconClicked from '../assets/watchlist-icon-clicked.svg';
+import WatchedIcon from '../assets/watched-icon.svg';
+import WatchedIconClicked from '../assets/watched-icon-clicked.svg';
+
+const Container = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-evenly;
+  padding-bottom: 10px;
+  margin: 5px 10px;
+`;
+
+const WatchListCheckLabel = styled.label`
+  display: flex;
+  flex-flow: column nowrap;
+  align-items: center;
+  margin: 10px;
+  padding: 5px;
+  cursor: pointer;
+
+  & :focus-within {
+    outline-width: 0;
+    border: 1px solid #aeb2f5;
+    border-radius: 8px;
+  }
+
+  & > * {
+    margin: 2.5px 0px;
+  }
+`;
+
+const WatchListCheck = styled.input`
+  position: absolute;
+  opacity: 0;
+  cursor: pointer;
+  height: 0;
+  width: 0;
+  user-select: none;
+`;
+
+const WatchListCheckText = styled.div`
+  display: block;
+  font-style: ${(props) => (props.checked ? 'italic' : 'normal')};
+  font-size: 0.7rem;
+  font-weight: 200;
+  font-family: 'Roboto', sans-serif;
+  color: #e7eaff;
+  background: transparent;
+`;
+
+function WatchlistButtons() {
+  const [watchlistAction, setWatchlistAction] = useState(null);
+
+  const addedToWatchlist = watchlistAction === 'addToWatchlist';
+  const addedToWatched = watchlistAction === 'addToWatched';
+
+  function handleWatchlistActionClick(event) {
+    const targetWatchlistAction = event.target.value;
+
+    if (watchlistAction === targetWatchlistAction) {
+      setWatchlistAction(null);
+    } else {
+      setWatchlistAction(targetWatchlistAction);
+    }
+  }
```

Comment:
I'd have two different handlers for each button, since you'll have to add more effects to the click later anyway. A `console.log('this will do a fetch, mark as watched and remove show from watchlist')` or something like that maybe?

Reply:
like this ? (see new commit)
I am not sure how to handle the actual functionality with the toggle function later. I need to do the fetch with an update to the lists when they close the modal since it would be too many requests if the user toggles it on and off multiple times...